src/sentry/static/sentry/app/views/eventsV2/table/queryField.tsx

```diff
@@ -40,9 +40,17 @@ type Props = {
   fieldOptions: FieldOptions;
   /**
    * The number of columns to render. Columns that do not have a parameter will
-   * render an empty parameter placeholder.
+   * render an empty parameter placeholder. Leave blank to avoid adding spacers.
    */
-  gridColumns: number;
+  gridColumns?: number;
+  /**
+   * Filter the options in the primary selector. Useful if you only want to
+   * show a subset of selectable items.
+   *
+   * NOTE: This is different from passing an already filtered fieldOptions
+   * list, as tag items in the list may be used as parameters to functions.
+   */
+  filterPrimaryOptions?: (option: SelectValue<FieldValue>) => boolean;
```

Comment:
The component has learned to filter it's primary options, but may still show other tags etc in the parameters.